### ISSUE TITLE
[accessibility issue] Icons appear in the accessibility tree

### ISSUE THREAD

**Issue by ckbedwell** [grafana/grafana#79513]
**Steps to reproduce**:
Go to a Grafana instance (e.g. play.grafana.com), open your dev tools and inspect the accessibility tree. Notice that all instances of `<Icon />` appear as an image with no description.

![A screenshot of a Grafana homepage with Chrome dev tools open on the accessibility tree and with VoiceOver rotor menu for images present. It shows that almost all of the images recognised by the screen rotor are just listed as "image". The image values in the accessibility tree are also blank.](https://github.com/grafana/grafana/assets/6906380/37625ba3-4c40-4a6f-bbb0-145312045b1e)

**Expected Result**
Most (all?) of our icons are purely decorative and don't add any extra meaningful information which isn't already provided by the surrounding context (e.g. an IconButton should have an aria-label on it).

I am slightly hesitant to suggest putting `aria-hidden="true"` on by default but I think this is probably a sensible default that developers can choose to opt out of?

An example is the keyboard icon in the command bar could add some useful text to benefit to indicate it is denoting a keyboard shortcut but is that implicit anyway by the accompanying text (cmd + k)?

UHG checkpoint: 1.111
